src/handlers/nedb-handler.js

```diff
@@ -0,0 +1,156 @@
+const DBInterface = require('./db-interface');
+const path        = require('path');
+const assert      = require('assert');
+const database    = require('nedb');
+const discordjs   = require('discord.js');
+
+// decent introduction here http://stackabuse.com/nedb-a-lightweight-javascript-database/
+// I think a real MongoDB solution would be overkill
+class NeDBHandler extends DBInterface {
+    constructor(bot, folder) {
+        super(bot);
+        this.folder = folder;
+        // This could have all be done in just one database file but I kept it seperated because that is what I think you did with your db
+        // also it makes it more human readable and gives access to the servernames mapped to their ids (could store more ofc)... so that's a plus
+        // load guild database that lists all the guilds that have a database - could also change that to just load all files in the guilds folder
+        this.guilds = new database(({filename: path.resolve(folder, 'guilds.json'), autoload: true}));
+        this.user   = new database(({filename: path.resolve(folder, 'users.json'), autoload: true}));
+        // load all guild databases - this could be optimized to only load a guild in when they are required... and not in the constructor...
+        let db      = new Map();
+        this.guilds.find({}, function (err, guilds) {
+            guilds.map(guild => {
+                db.set(guild.id, new database(({filename: path.resolve(folder, 'guilds', guild.id + '.json'), autoload: true})));
+            });
+        });
+        this.db = db;
+    }
+
+    getDBHandle(guild) {
+        if (!(guild instanceof discordjs.Guild)) { throw 'dbHandle must be passed a guild!'; }
+        let guild_db = this.db.get(guild.id);
+        if (!guild_db) { // if guild doesn't has a database already create one, add the server and return the db of the guild
+            guild_db = new database(({filename: path.resolve(this.folder, 'guilds', guild.id + '.json'), autoload: true}));
+            this.db.set(guild.id, guild_db);
+            this.guilds.insert({id: guild.id, name: guild.name,});
+        }
+        return guild_db;
+    }
+
+    /**
+     * Get a value for a key for a database
+     * @param guild
+     * @param key
+     * @returns {Promise} resolves into the value or null
+     */
+    get(guild, key) {
```

Comment:
This should be async, right? The return line would need to be modified to appropriately work too.